packages/key-readmodel-writer/src/keyConsumerServiceV1.ts

```diff
@@ -1,15 +1,47 @@
 import { KeyCollection } from "pagopa-interop-commons";
-import { AuthorizationEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  AuthorizationEventEnvelopeV1,
+  fromKeyV1,
+  toReadModelKey,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
 
 export async function handleMessageV1(
   message: AuthorizationEventEnvelopeV1,
-  _keys: KeyCollection
+  keys: KeyCollection
 ): Promise<void> {
-  match(message)
+  await match(message)
+    .with({ type: "KeysAdded" }, async (message) => {
+      const keysToAdd = message.data.keys.map((keyEntry) => keyEntry.value);
+
+      for (const key of keysToAdd) {
+        if (key) {
+          await keys.updateOne(
+            {
+              "data.kid": key.kid,
+              "metadata.version": { $lt: message.version },
+            },
+            {
+              $set: {
+                data: toReadModelKey(fromKeyV1(key)),
+                metadata: {
+                  version: message.version,
+                },
+              },
+            },
+            { upsert: true }
+          );
+        }
+        await Promise.resolve();
+      }
+    })
+    .with({ type: "KeyDeleted" }, async (message) => {
+      await keys.deleteOne({
+        "data.kid": message.data.keyId,
+        "metadata.version": { $lt: message.version },
+      });
+    })
     .with(
-      { type: "KeysAdded" },
-      { type: "KeyDeleted" },
       { type: "KeyRelationshipToUserMigrated" },
```

Comment:
`KeyRelationshipToUserMigrated` has been used to replace relationship ids with user ids for all existing keys.
This event should be handled in order to add `userId` field for keys created before that.

Reply:
Updated